### PR TITLE
Objectwrap and ClassPropertyDescriptor  documentation

### PR DESCRIPTION
**Pull request by NickNaso**
Hi everyone,
I just start on writing documentation for `Napi::ObjectWrap` please review and suggest me some changes and possible improvements.

### CODE REVIEW

File: doc/object_wrap.md [hunk 1]
```diff
@@ -1,13 +1,384 @@
...
+
+Napi::Object Example::Init(Napi::Env env, Napi::Object exports) {
+    Napi::HandleScope scope(env);
+    // This method is used to hook the accessor and method callbacks
+    Napi::Function func = DefineClass(env, "Example", {
+        InstanceMethod("GetValue", &Example::GetValue),
+        InstanceMethod("SetValue", &Example::SetValue)
+    });
+
+    constructor = Napi::Persistent(func);
```
Comment (mhdawson) [inline doc/object_wrap.md]: Some comments here that talk about how the persistent is used and why suppress destruct is needed would be good.

File: doc/object_wrap.md [hunk 2]
```diff
@@ -1,13 +1,384 @@
...
+
+  private:
+    static Napi::FunctionReference constructor;
+    double _value;
+    Napi::Value GetValue(const Napi::CallbackInfo &info);
+    Napi::Value SetValue(const Napi::CallbackInfo &info);
+};
+
+Napi::Object Example::Init(Napi::Env env, Napi::Object exports) {
+    Napi::HandleScope scope(env);
```
Comment (mhdawson) [inline doc/object_wrap.md]: Is the handle scope needed here? I'm not sure but worth checking

**Comment by mhdawson** [comment]
Ran out of time today, will try to add more comments tomorrow.

File: doc/object_wrap.md (hunk 1 above)
Comment (NickNaso) [reply doc/object_wrap.md]: No it's not necessary I will remove it. (I tested the example without handle scope).

File: doc/object_wrap.md (hunk 2 above)
Comment (NickNaso) [reply doc/object_wrap.md]: No it's not necessary I will remove it. (I tested the example without handle scope).

File: doc/object_wrap.md (hunk 1 above)
Comment (NickNaso) [reply doc/object_wrap.md]: Ok it's reasonable give more explanation about this.

**Comment by mhdawson** [comment]
Landed as 100d0a7cb20de705f71b77be68f2dc2075ec13ae

**Pull request closed without merging**